celery/worker/components.py

```diff
@@ -95,6 +95,47 @@ def _patch_thread_primitives(self, w):
             pool.Lock = DummyLock
 
 
+class Hub2(bootsteps.StartStopStep):
+    """Worker starts the event loop."""
+
+    requires = (Timer,)
+
+    def __init__(self, w, **kwargs):
+        w.hub2 = None
+        super().__init__(w, **kwargs)
+
+    def include_if(self, w):
+        return w.use_eventloop
+
+    def create(self, w):
+        if w.hub2 is None:
+            required_hub = getattr(w._conninfo, 'requires_hub', None)
+            w.hub2 = set_event_loop((
```

Comment:
Shouldn’t use the `set_event_loop` because it uses the global. It needs to create a new instance.

Reply:
It needs to create a hub per consumer (e.g. from `w.consumers`)

Reply:
Integrate logic into Hub class:
1. If `w.consumer` is NOT none, go as usual.
2. if `w.consumers` is NOT none, apply same logic for each consumer (to get its own hub (loop) instance).